etc/settings.yaml.in

```diff
@@ -1,25 +1,66 @@
 locations:
-  input-dir: in/
-  output-dir: out/
-  genome-dir: genome/
+  input-dir: /path/to/reads/
+  output-dir: /path/to/output/
+  genome-fasta: /path/to/genome.fasta
 
 general:
-  assembly: hg19
+  assembly: ''
+  # use one or both bisulfite aligners
+  # bwameth is faster and uses less resources
+  use_bwameth: False
+  # bismark is gold standard and uses sensible defaults
+  use_bismark: True
   methylation-calling:
     minimum-coverage: 10
     minimum-quality: 10
+    # this applies to bwameth only
+    keep-singleton: False
+    # keep-dups can be Auto/True/False, Auto decides based on protocol
+    keep-Dups: Auto                 
+  export-bigwig:
+    # extract/export contexts other than CpG? 
+    context:
+      cpg: 
+        export: True
+        # join both strands and summarize coverage/methylation? 
+        destrand: True
+      chg: 
+        export: False
+        destrand: False
+      chh: 
+        export: False
+        destrand: False
   reports:
     TSS_plotlength: 5000
   differential-methylation:
     cores: 1
     qvalue: 0.01
     difference: 25
-    treatment-groups:
-      - []
     annotation:
-      cpgIsland_bedfile: "pigx_work/refGenome/cpgIslandExt.hg19.bed.gz"
-      refGenes_bedfile:  "pigx_work/refGenome/refGene.hg19.bed.gz"
+      cpgIsland-bedfile: ''
+      refGenes-bedfile: ''
+      # download cpgIsland-bedfile or refGenes-bedfile automatically if not given?
       webfetch:   no
+      
+# DManalyses:
+#   # The names of analyses can be anything but they have to be unique
+#   # for each combination of case control group comparisons.
+
+#   analysis1:
+#     treatment_sample_groups: "MED1"
+#     control_sample_groups: "H2O"
+
+
+#  multipleTreat:
+#     # If multiple sample groups are provided, they must be separated by comma.
+#    treatment_sample_groups: "MED2,MED1"
+#    control_sample_groups: "H2O"
+
+#  withinGroup:
+#    # For within group comparison (all vs all), give the same groups to treatment and control.
+#    treatment_sample_groups: "MED2,MED1,H2O"
```

Comment:
it's better to use arrays than comma-separated strings whenever we want to have multiple values.  Same applies above and below.